src/Views/Gestures.vala

```diff
@@ -0,0 +1,251 @@
+        glib_settings.bind ("multitasking-gesture-enabled", multitasking_switch, "active", GLib.SettingsBindFlags.DEFAULT);
+        multitasking_switch.bind_property ("active", multitasking_combobox, "sensitive", BindingFlags.SYNC_CREATE);
+        multitasking_combobox.changed.connect (() => update_comboboxes (multitasking_combobox));
+
+        glib_settings.bind ("workspaces-gesture-enabled", workspaces_switch, "active", GLib.SettingsBindFlags.DEFAULT);
+        workspaces_switch.bind_property ("active", workspaces_combobox, "sensitive", BindingFlags.SYNC_CREATE);
+        workspaces_combobox.changed.connect (() => update_comboboxes (workspaces_combobox));
+
+        maximize_switch.bind_property ("active", maximize_combobox, "sensitive", BindingFlags.SYNC_CREATE);
+        maximize_combobox.changed.connect (() => update_comboboxes (maximize_combobox));
+        maximize_switch.state_set.connect (() => {
+            int fingers = int.parse (maximize_combobox.get_active_id ());
+            save_combobox_settings (maximize_combobox, fingers);
+            update_ui ();
+        });
+
+        tile_switch.bind_property ("active", tile_combobox, "sensitive", BindingFlags.SYNC_CREATE);
+        tile_combobox.changed.connect (() => update_comboboxes (tile_combobox));
+        tile_switch.state_set.connect (() => {
+            int fingers = int.parse (tile_combobox.get_active_id ());
+            save_combobox_settings (tile_combobox, fingers);
+            update_ui ();
+        });
+    }
+
+    private void update_comboboxes (Gtk.ComboBoxText combobox) {
+        string selection = combobox.get_active_id ();
+        int fingers = int.parse (selection);
+
+        var linked_combobox = get_linked_combobox (combobox);
+        var linked_fingers = (fingers == 3) ? 4 : 3;
+
+        save_combobox_settings (combobox, fingers);
+        if (linked_combobox.get_active_id () == selection) {
+            save_combobox_settings (linked_combobox, linked_fingers);
+        }
+
+        update_ui ();
+    }
+
+    private Gtk.ComboBoxText? get_linked_combobox (Gtk.ComboBoxText combobox) {
+        if (combobox == multitasking_combobox) {
+            return maximize_combobox;
+        }
+
+        if (combobox == workspaces_combobox) {
+            return tile_combobox;
+        }
+
+        if (combobox == maximize_combobox) {
+            return multitasking_combobox;
+        }
+
+        if (combobox == tile_combobox) {
+            return workspaces_combobox;
+        }
+
+        // Unreachable
+        return null;
+    }
+
+    private void save_combobox_settings (Gtk.ComboBoxText combobox, int fingers) {
+        if (combobox == multitasking_combobox) {
+            glib_settings.set_int ("multitasking-gesture-fingers", fingers);
+            return;
+        }
+
+        if (combobox == workspaces_combobox) {
+            glib_settings.set_int ("workspaces-gesture-fingers", fingers);
+            return;
+        }
```

Comment:
Instead of this, have you tried binding `active_id`?

Reply:
Even though I can do it with this 2 comboboxes, I can't with maximize and tile... Or at least I didn't find a way. This is my first time with GTK, so most likely, I'm missing something.
I like it more this way, because it allows me to reuse the logic inside `save_combobox_settings`.